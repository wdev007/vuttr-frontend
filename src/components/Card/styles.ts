import styled from 'styled-components';

export const CardContainer = styled.div`
  height: 145px;
  width: 100%;

  background: #ffffff 0% 0% no-repeat padding-box;
  border: 1px solid #ebeaed;
  border-radius: 5px;
  transition: transform 0.2s;

  & + div {
    margin-top: 10px;
  }

  &:hover {
    transform: translateX(5px);
    box-shadow: 0px 5px 7px #0000000d;
  }

  div {
    display: flex;
    justify-content: space-between;

    margin: 5px 5px 0;
  }
`;

export const ButtonRemove = styled.button`
  display: flex;
  align-items: flex-end;
  justify-content: center;

  width: 100px;
  height: 40px;
  background: #f95e5a 0% 0% no-repeat padding-box;
  color: #fff;
  border: none;
  border-radius: 5px;

  &:hover {
    background: #cc4c4c 0% 0% no-repeat padding-box;
  }
  &:active {
    background: #a53f3f 0% 0% no-repeat padding-box;
  }
`;
