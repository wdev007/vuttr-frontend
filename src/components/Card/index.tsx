import React from 'react';
import { AiOutlineClose } from 'react-icons/ai';

import { Tool } from '../../models/tool';
import { CardContainer, ButtonRemove } from './styles';

const CardComponent: React.FC<Tool> = ({
  id,
  title,
  link = '',
  description,
  tags,
}) => (
  <CardContainer id={id}>
    <div>
      <a href={link}>{title}</a>
      <ButtonRemove type="button">
        <AiOutlineClose /> remove
      </ButtonRemove>
    </div>
    <p>{description}</p>
    {tags.length && tags?.map(tag => <strong key={tag}>#{tag}</strong>)}
  </CardContainer>
);

export default CardComponent;
