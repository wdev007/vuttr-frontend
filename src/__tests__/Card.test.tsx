import React from 'react';
import { shallow } from 'enzyme';

import Card from '../components/Card';

describe('Card', () => {
  const props = {
    id: 'Notion',
    title: 'Notion',
    tags: ['organization', 'planning', 'collaboration', 'writing', 'calendar'],
    link: 'https://notion.so',
    description:
      'All in one tool to organize teams and ideas. Write, plan, collaborate, and get organized. ',
  };

  it('should render correctly with no props', () => {
    const component = shallow(<Card {...props} />);

    expect(component).toMatchSnapshot();
  });
});
