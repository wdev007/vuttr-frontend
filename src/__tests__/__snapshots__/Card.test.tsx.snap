// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly with no props 1`] = `
<Fragment>
  <styled.div
    id="Notion"
  >
    <div
      className="actions-container"
    >
      <styled.a
        className=""
        href="https://notion.so"
        id="card-title"
      >
        Notion
      </styled.a>
      <styled.button
        onClick={[Function]}
        type="button"
      >
        <AiOutlineClose />
         remove
      </styled.button>
    </div>
    <p>
      All in one tool to organize teams and ideas. Write, plan, collaborate, and get organized. 
    </p>
    <div
      className="tags-container"
    >
      <strong
        className=""
        key="organization"
      >
        #
        organization
      </strong>
      <strong
        className=""
        key="planning"
      >
        #
        planning
      </strong>
      <strong
        className=""
        key="collaboration"
      >
        #
        collaboration
      </strong>
      <strong
        className=""
        key="writing"
      >
        #
        writing
      </strong>
      <strong
        className=""
        key="calendar"
      >
        #
        calendar
      </strong>
    </div>
  </styled.div>
  <ModalComponent
    cancelText="Cancel"
    iconTitle={[Function]}
    okText="Yes, remove"
    onCancel={[Function]}
    onOk={[Function]}
    title="Remove tool"
    visible={false}
  >
    <p>
      Are you sure you want to remove
      <strong
        className="text-negrito"
      >
        Notion
      </strong>
       ?
    </p>
  </ModalComponent>
</Fragment>
`;
